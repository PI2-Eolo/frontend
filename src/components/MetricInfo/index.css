.metric-info-container {
    display: flex;
    flex-direction: column;
    width: fit-content;
}

.value-container {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    justify-content: center;
}

.label {
    font-family: "Poppins-Thin";
    font-size: 1em;
    font-weight: 600;
}

.value {
    font-family: "Poppins-Thin";
    font-weight: 100;
    font-size: 3em;
    margin-right: 0.2em;
    margin-left: 0.5em;
}

.unit {
    font-family: "Poppins-Thin";
    font-weight: 100;
    font-size: 0.9em;
    margin-bottom: 0.6em;
}