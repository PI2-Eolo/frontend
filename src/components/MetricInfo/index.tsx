import React from "react";
import "./index.css";

interface MetricInforProps {
  label: string;
  value: number;
  unit: string;
  decimalPlaces?: number;
}

const MetricInfo = ({
  label,
  value,
  unit,
  decimalPlaces = 2,
}: MetricInforProps) => {
  const stringValue = value.toFixed(decimalPlaces);

  return (
    <div className="metric-info-container">
      <span className="label">{label}</span>
      <div className="value-container">
        <span className="value">{stringValue}</span>
        <span className="unit">{unit}</span>
      </div>
    </div>
  );
};

export default MetricInfo;
