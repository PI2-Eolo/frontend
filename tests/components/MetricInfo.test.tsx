import React from 'react';
import renderer from 'react-test-renderer';
import MetricInfo from '../../src/components/MetricInfo';

describe('Header component test', () => {
    it('component snapshot', () => {
        const consumptionComponent = renderer.create(
            <MetricInfo label="Consumption" value={13.3} unit="kW/h"/>,
        );
        expect(consumptionComponent.toJSON()).toMatchSnapshot();
        const energyComponent = renderer.create(
            <MetricInfo label="Energy production" value={25.1} unit="kw/h" decimalPlaces={3}/>
        );
        expect(energyComponent.toJSON()).toMatchSnapshot();
    })
})