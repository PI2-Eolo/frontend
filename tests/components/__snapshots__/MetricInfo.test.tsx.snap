// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test component snapshot 1`] = `
<div
  className="metric-info-container"
>
  <span
    className="label"
  >
    Consumption
  </span>
  <div
    className="value-container"
  >
    <span
      className="value"
    >
      13.30
    </span>
    <span
      className="unit"
    >
      kW/h
    </span>
  </div>
</div>
`;

exports[`Header component test component snapshot 2`] = `
<div
  className="metric-info-container"
>
  <span
    className="label"
  >
    Energy production
  </span>
  <div
    className="value-container"
  >
    <span
      className="value"
    >
      25.100
    </span>
    <span
      className="unit"
    >
      kw/h
    </span>
  </div>
</div>
`;
